File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsRunForeverTask.java
package org.murraybridgebunyips.bunyipslib.tasks.bases;

import androidx.annotation.NonNull;

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode;
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem;

/**
 * RunForeverTask interface with BunyipsOpMode dependency injection.
 */
public abstract class BunyipsRunForeverTask extends RunForeverTask {
    protected final BunyipsOpMode opMode;

    protected BunyipsRunForeverTask(BunyipsOpMode opMode) {
        this.opMode = opMode;
    }

    protected BunyipsRunForeverTask(BunyipsOpMode opMode, @NonNull BunyipsSubsystem dependencySubsystem, boolean shouldOverrideConflictingTasks) {
        super(dependencySubsystem, shouldOverrideConflictingTasks);
        this.opMode = opMode;
    }
}

File: TeamCode/src/org/murraybridgebunyips/bunyipslib/tasks/bases/BunyipsRunForeverTask.kt
package org.murraybridgebunyips.bunyipslib.tasks.bases

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode
import org.murraybridgebunyips.bunyipslib.BunyipsSubsystem

/**
 * RunForeverTask interface with BunyipsOpMode dependency injection.
 */
abstract class BunyipsRunForeverTask(open val opMode: BunyipsOpMode) : RunForeverTask() {
    constructor(
        opMode: BunyipsOpMode,
        dependencySubsystem: BunyipsSubsystem,
        shouldOverrideConflictingTasks: Boolean
    ) : this(opMode) {
        addDependency(dependencySubsystem)
        overrideOnConflict = shouldOverrideConflictingTasks
    }
}
